python/tank/util/version.py

```diff
@@ -7,34 +7,70 @@
 # By accessing, using, copying or modifying this work you indicate your
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
-import warnings
 import contextlib
 import sys
+import warnings
 
-LooseVersion = None
+version_parse = None
 try:
     import packaging.version
+
+    version_parse = packaging.version.parse
 except ModuleNotFoundError:
     try:
         # Try importing from setuptools.
         # If it fails, then we can't do much at the moment
         # The DCC should have either setuptools or packaging installed.
         from setuptools._distutils.version import LooseVersion
+
+        version_parse = LooseVersion
     except ModuleNotFoundError:
         try:
-            # DCCs with older versions of Python 3.12
             from distutils.version import LooseVersion
+
+            version_parse = LooseVersion
         except ModuleNotFoundError:
-            pass
+            warnings.warn(
+                "No version parsing library available (packaging, setuptools, or distutils). "
+                "Falling back to string comparison which may produce incorrect version ordering.",
+                UserWarning,
+            )
+            version_parse = str
 
-from . import sgre as re
 from .. import LogManager
 from ..errors import TankError
-
+from . import sgre as re
 
 logger = LogManager.get_logger(__name__)
 GITHUB_HASH_RE = re.compile("^[0-9a-fA-F]{7,40}$")
 
+# Normalize non-standard version formats
+# into PEP 440–compliant forms ("1.2.3") to ensure compatibility with
+# Python’s version parsing utilities (e.g., packaging.version.parse).
+# Reference: https://peps.python.org/pep-0440/
+_VERSION_PATTERNS = [
+    (
+        re.compile(r"^[a-zA-Z\s]+(\d+(?:\.\d+)*)(?:\s|$)"),
+        r"\1",
+    ),  # Extract version from software names: "Software Name 21.0" -> "21.0"
+    (
+        re.compile(r"^(\d+)\.(\d+)v(\d+)$"),
+        r"\1.\2.\3",
+    ),  # Dot-v format: "6.3v6" -> "6.3.6"
+    (
+        re.compile(r"^(\d+)v(\d+(?:\.\d+)*)$"),
+        r"\1.\2",
+    ),  # Simple v format: "2019v0.1" -> "2019.0.1"
+    (
+        re.compile(r"^(\d+(?:\.\d+)*)(sp|hotfix|hf)(\d+)$"),
+        r"\1.post\3",
+    ),  # Service pack without dot: "2017.2sp1" -> "2017.2.post1"
+    (
+        re.compile(r"^(\d+(?:\.\d+)*)\.(sp|hotfix|hf)(\d+)$"),
+        r"\1.post\3",
+    ),  # Service pack with dot: "2017.2.sp1" -> "2017.2.post1"
```

Comment:
Since the difference between the two is only a dot. We can make the dot optional no? So having only one regexp handling with or withoug dots (2 examples)